### PR TITLE
[Bug 16300] Resolve correctly serial port paths on Windows

### PR DESCRIPTION
**Pull request by livecodepanos**

### CODE REVIEW

**Comment by peter-b** [comment]
@livecodepanos Do you think it would be a good idea to make this stricter by checking that all of the characters between the "COM" and the ":" are digits?  This would be an ideal application for a regular expression -- you could use `MCStringsEvalMatchText()` here quite easily I think.

**Comment by runrevmark** [comment]
I think an RE would be overkill here... The 'reserved' device names on Win32 are:

CON, PRN, AUX, NUL, COM1, COM2, COM3, COM4, COM5, COM6, COM7, COM8, COM9, LPT1, LPT2, LPT3, LPT4, LPT5, LPT6, LPT7, LPT8, and LPT9

I believe - so maybe factor into a separate function to check for all of those... You just need to check the first 3 chars, then for a digit (if necessary) then for a colon.

**Comment by peter-b** [comment]
@runrevmark Unfortunately I've come across real world Windows systems with COM15 ports (esp. when using USB-Serial adapters). If this code path treats "COM15:" as a regular filename, will it still be possible to open COM15 in LiveCode?

File: engine/src/dskw32.cpp [hunk 1]
```diff
@@ -3697,6 +3755,13 @@ struct MCWindowsDesktop: public MCSystemInterface, public MCWindowsSystemService
 			// absolute path
 			t_canonised_path = p_path;
 		}
+		
+		// PM-2016-03-15: [[ Bug 16300 ]] Detect correctly the case of a serial port path
+		else if (isValidSerialPortPath(MCStringRef p_path))
+		{
+			t_canonised_path = MCValueRetain(p_path);
```
Comment (livecodeali) [inline engine/src/dskw32.cpp]: This shouldn't be `MCValueRetain`ed - something assigned to an AutoValueRef is automatically retained.

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 5b456ab

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 5b456ab284157a118018c4802f89a32939814ffc

**Comment by livecode-vulcan** [comment]
😞 test failure 5b456ab284157a118018c4802f89a32939814ffc
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2181)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2189)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2174)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2190)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2196)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2201)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2212)

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 6056c35

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 6056c3544034d3a06fa3a1ca147afe94847dd303

**Comment by livecode-vulcan** [comment]
😎 test success 6056c3544034d3a06fa3a1ca147afe94847dd303
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2187)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2194)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2180)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2196)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2201)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2207)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2218)

**Pull request merged**